ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -171,6 +173,7 @@ private BlobStore(ReplicaId replicaId, String storeId, StoreConfig config, Sched
     this.thresholdBytesLow = (long) (capacityInBytes * ((threshold - delta) / 100.0));
     ttlUpdateBufferTimeMs = TimeUnit.SECONDS.toMillis(config.storeTtlUpdateBufferTimeSeconds);
     errorCount = new AtomicInteger(0);
+    currentState = ReplicaState.OFFLINE;
```

Comment:
I don't see any callers to `setCurrentState()` in the production code. Will this be added in a future PR?

Reply:
Right, this will be called in storage manager when state transition occurs. I was planning to make the changes in this PR but then decided to split it into a separate PR.